pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -641,7 +641,7 @@ contract ManagedPool is BaseWeightedPool, ProtocolFeeCache, ReentrancyGuard, ICo
         // As we're working with normalized weights, `totalWeight` is equal to 1.
         //
         // We can then easily calculate the new denormalized weight sum by applying this ratio to the old sum.
-        uint256 weightSumAfterAdd = _denormWeightSum.mulUp(FixedPoint.ONE.divDown(FixedPoint.ONE - normalizedWeight));
+        uint256 weightSumAfterAdd = _denormWeightSum.divDown(FixedPoint.ONE - normalizedWeight);
```

Comment:
Was this just wrong?

Reply:
`x.mulUp(FixedPoint.ONE) == x`

It matches the maths but we can just remove this and get the same result.

Reply:
```
// Before (I'm replacing mulUp with mulDown to ignore rounding)
weightSumAfterAdd = _denormWeightSum.mulUp(FixedPoint.ONE.divDown(FixedPoint.ONE - normalizedWeight));
weightSumAfterAdd = _denormWeightSum * (ONE * ONE / (ONE - normalizedWeight)  ) / ONE
weightSumAfterAdd = _denormWeightSum * ( ONE / (ONE - normalizedWeight)  )

// After
weightSumAfterAdd = _denormWeightSum.divDown(FixedPoint.ONE - normalizedWeight);
weightSumAfterAdd = _denormWeightSum * ONE / (ONE - normalizedWeight)
```

So these are equivalent up to a rounding error.